### PR TITLE
[MB-6392] Replace MomentJS with DayJS

### PR DESCRIPTION
**Pull request by suzubara**
## Description

This PR replaces our date/time parse & format library Moment.js with [Day.js](https://day.js.org/en/) (due to [the Moment.js project being deprecated](https://momentjs.com/docs/#/-project-status/)).

In _most_ cases this was a 1-1 swap out since Day.js largely uses the same API as Moment.js. For certain features (such as `fromNow()` and parsing from non-ISO formatted date strings), Day.js relies on being extended with plugins.

I did have to make one change where we use `react-day-picker`, because there was a slight discrepancy between how Day.js & Moment.js parse date strings. Since the `DayPickerInput` is always working with a parsed Date object for its value, the easiest way to reconcile this was to provide it with a `formatDate` function that _always_ formats a date from ISO (instead of a custom format).

I also deleted any date/time formatter functions that were not being used or referenced anywhere. There's 💯  some more organization that could be done around all of the various date formatting code we have, but I limited the scope to avoid making this PR bigger than it needed to be.

Because Moment was listed under Cypress dependencies, there is a corresponding PR in our Cypress docker image repo: https://github.com/transcom/circleci-docker/pull/81
We were not actually using Moment in any Cypress code, so I removed it.

## Reviewer Notes

Tests passing are a good sign, but you can also go into the app and verify anywhere dates are displayed or forms that include a date picker input are still functioning as expected (verify initial date values still work, and you are able to select a new date value). Forms that include date picker inputs are:
- Orders Info (Customer onboarding)
- Create/Edit MTO Shipment (Customer onboarding)
- Orders Detail (Office)

## Setup

To run the app locally:

```sh
make server_run
make client_run
```

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6392) for this change

### CODE REVIEW

**Review by Shkeating — APPROVED** [review]
👍😎👍

**Comment by suzubara** [comment]
**UPDATE:** After investigating the integration test failures on this branch, I've deduced that these changes to our date parse/format functions are not working on date inputs that use the [SwaggerField](https://github.com/transcom/mymove/blob/master/src/shared/JsonSchemaForm/JsonSchemaField.js#L252) component (this includes the Customer PPM flow and Office PPM forms). `SwaggerField` is a legacy component that, due to being extremely reliant on configuration and itself used in 30+ forms to generate any kind of form input, I don't feel comfortable making significant edits to. It's also difficult (and would be tedious) to pinpoint all of the places where `SwaggerField` is used to generate date inputs, since this logic is driven by the Swagger API schema rather than declarative UI code.

Given that we are planning to replace usage of `SwaggerField` with more explicit forms built using Formik and ReactUSWDS components, it doesn't seem worth trying to debug and resolve a component that could have sweeping effects in places of the app that have relatively low test coverage. Given all of that, I am recommending we put this work on hold until after `SwaggerField` is deprecated. MomentJS has been deprecated, but there are no immediate known issues resulting from us using it, so I don't believe there is any risk to us delaying this work.

File: src/components/Office/OrdersDetailForm/OrdersDetailForm.stories.jsx [hunk 1]
```diff
@@ -55,7 +55,20 @@ export default {
 };
 
 export const EmptyValues = () => (
-  <Formik>
+  <Formik
+    initialValues={{
```
Comment (jenniferlynparsons) [inline src/components/Office/OrdersDetailForm/OrdersDetailForm.stories.jsx:59]: Nice addition!!!

**Review by jenniferlynparsons — APPROVED** [review]
Looks great!

**Comment by suzubara** [comment]
Closing this for the meantime since the way PPM Office uses date inputs is not working with DayJS

**Pull request closed without merging**